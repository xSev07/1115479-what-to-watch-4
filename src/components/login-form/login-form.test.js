import React from "react";
import renderer from "react-test-renderer";
import LoginForm from "./login-form";

describe(`Should LoginForm render correctly`, () => {
  it(`should render basic form`, () => {
    const tree = renderer
      .create(
          <LoginForm
            authError={false}
            incorrectEmail={false}
            incorrectPassword={false}
            onSubmit={() => {}}
          />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`should render form with auth error`, () => {
    const tree = renderer
      .create(
          <LoginForm
            authError={true}
            incorrectEmail={false}
            incorrectPassword={false}
            onSubmit={() => {}}
          />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`should render form with incorrect email`, () => {
    const tree = renderer
      .create(
          <LoginForm
            authError={false}
            incorrectEmail={true}
            incorrectPassword={false}
            onSubmit={() => {}}
          />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`should render form with incorrect password`, () => {
    const tree = renderer
      .create(
          <LoginForm
            authError={false}
            incorrectEmail={false}
            incorrectPassword={true}
            onSubmit={() => {}}
          />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
