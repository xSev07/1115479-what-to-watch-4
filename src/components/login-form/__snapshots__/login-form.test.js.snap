// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should LoginForm render correctly should render basic form 1`] = `
<form
  action="#"
  className="sign-in__form"
>
  <div
    className="sign-in__fields"
  >
    <div
      className="sign-in__field"
    >
      <input
        className="sign-in__input"
        id="user-email"
        name="email"
        placeholder="Email address"
        type="email"
      />
      <label
        className="sign-in__label visually-hidden"
        htmlFor="user-email"
      >
        Email address
      </label>
    </div>
    <div
      className="sign-in__field"
    >
      <input
        className="sign-in__input"
        id="user-password"
        name="password"
        placeholder="Password"
        type="password"
      />
      <label
        className="sign-in__label visually-hidden"
        htmlFor="user-password"
      >
        Password
      </label>
    </div>
  </div>
  <div
    className="sign-in__submit"
  >
    <button
      className="sign-in__btn"
      onClick={[Function]}
      type="submit"
    >
      Sign in
    </button>
  </div>
</form>
`;

exports[`Should LoginForm render correctly should render form with auth error 1`] = `
<form
  action="#"
  className="sign-in__form"
>
  <div
    className="sign-in__message"
  >
    <p>
      We canâ€™t recognize this email 
      <br />
       and password combination. Please try again.
    </p>
  </div>
  <div
    className="sign-in__fields"
  >
    <div
      className="sign-in__field"
    >
      <input
        className="sign-in__input"
        id="user-email"
        name="email"
        placeholder="Email address"
        type="email"
      />
      <label
        className="sign-in__label visually-hidden"
        htmlFor="user-email"
      >
        Email address
      </label>
    </div>
    <div
      className="sign-in__field"
    >
      <input
        className="sign-in__input"
        id="user-password"
        name="password"
        placeholder="Password"
        type="password"
      />
      <label
        className="sign-in__label visually-hidden"
        htmlFor="user-password"
      >
        Password
      </label>
    </div>
  </div>
  <div
    className="sign-in__submit"
  >
    <button
      className="sign-in__btn"
      onClick={[Function]}
      type="submit"
    >
      Sign in
    </button>
  </div>
</form>
`;

exports[`Should LoginForm render correctly should render form with incorrect email 1`] = `
<form
  action="#"
  className="sign-in__form"
>
  <div
    className="sign-in__message"
  >
    <p>
      Please enter a valid user data
    </p>
  </div>
  <div
    className="sign-in__fields"
  >
    <div
      className="sign-in__field sign-in__field--error"
    >
      <input
        className="sign-in__input"
        id="user-email"
        name="email"
        placeholder="Email address"
        type="email"
      />
      <label
        className="sign-in__label visually-hidden"
        htmlFor="user-email"
      >
        Email address
      </label>
    </div>
    <div
      className="sign-in__field"
    >
      <input
        className="sign-in__input"
        id="user-password"
        name="password"
        placeholder="Password"
        type="password"
      />
      <label
        className="sign-in__label visually-hidden"
        htmlFor="user-password"
      >
        Password
      </label>
    </div>
  </div>
  <div
    className="sign-in__submit"
  >
    <button
      className="sign-in__btn"
      onClick={[Function]}
      type="submit"
    >
      Sign in
    </button>
  </div>
</form>
`;

exports[`Should LoginForm render correctly should render form with incorrect password 1`] = `
<form
  action="#"
  className="sign-in__form"
>
  <div
    className="sign-in__message"
  >
    <p>
      Please enter a valid user data
    </p>
  </div>
  <div
    className="sign-in__fields"
  >
    <div
      className="sign-in__field"
    >
      <input
        className="sign-in__input"
        id="user-email"
        name="email"
        placeholder="Email address"
        type="email"
      />
      <label
        className="sign-in__label visually-hidden"
        htmlFor="user-email"
      >
        Email address
      </label>
    </div>
    <div
      className="sign-in__field sign-in__field--error"
    >
      <input
        className="sign-in__input"
        id="user-password"
        name="password"
        placeholder="Password"
        type="password"
      />
      <label
        className="sign-in__label visually-hidden"
        htmlFor="user-password"
      >
        Password
      </label>
    </div>
  </div>
  <div
    className="sign-in__submit"
  >
    <button
      className="sign-in__btn"
      onClick={[Function]}
      type="submit"
    >
      Sign in
    </button>
  </div>
</form>
`;
